### PR TITLE
[Frontend] Add support for Hamiltonian with integer coefficients

### PR DESCRIPTION
**Pull request by maliasadi**
**Context:**
Catalyst doesn’t support Hamiltonians with integer coefficients (`coeffs` with `int` element type) while this is supported in PennyLane. There are more use cases for this type of Hamiltonians in particular for `qml.sum` where they're converted to PL Hamiltonian data-types with integer coefficients.  (e.g., `qml.PauliZ(0) + qml.PauliZ(1)` is converted to `H([1,1], [Z(0), Z(1)])`).

This PR adds this support with updating the Hamiltonian primitive.

### CODE REVIEW

File: doc/changelog.md [hunk 1]
```diff
@@ -84,6 +84,24 @@
   array(112936.34906843)
   ```
 
+* Add support for Hamiltonian observables with integer coefficients.
+  [#248](https://github.com/PennyLaneAI/catalyst/pull/248)
+
+  For example, compiling the following circuit wasn't allowed before, but it is
+  not supported in Catalyst:
```
Comment (dime10) [inline doc/changelog.md:91]:
```suggestion
  now supported in Catalyst:

```

File: frontend/test/pytest/test_measurements_results.py
Comment (dime10) [inline frontend/test/pytest/test_measurements_results.py:1]: Just curious the inputs in some existing tests were changed here :)

**Review by dime10 — APPROVED** [review]
Thanks @maliasadi 💯

File: frontend/test/pytest/test_measurements_results.py
Comment (maliasadi) [reply frontend/test/pytest/test_measurements_results.py:1]: Only the indentation of these tests has changed. GH UI is failing to capture the changes properly

Comment (dime10) [reply frontend/test/pytest/test_measurements_results.py:1]: Good point 👍

**Pull request merged**